### ISSUE TITLE
  ModuleNotFoundError: No module named 'torch'

### ISSUE THREAD

**Issue by lucasjinreal** [Dao-AILab/flash-attention#253]
torch actually installed

**Comment by dstieglitz** [comment]
@xwyzsn I have the same problem as @lucasjinreal, tried your suggestions, and get "ModuleNotFoundError: No module named 'packaging'"

**Comment by vchiley** [comment]
this cant be done within a setup.py file in my project...

**Comment by REIGN12** [comment]
Maybe regressing to older version of flash-attn is sufficient. I succeed with flash-attn==1.0.4

**Comment by sytelus** [comment]
I'm getting below with `--no-build-isolation` as well as with 1.0.4.

```
ModuleNotFoundError: No module named 'torch.utils'
```

**Comment by sytelus** [comment]
> I'm getting below with `--no-build-isolation` as well as with 1.0.4.
> 
> ```
> ModuleNotFoundError: No module named 'torch.utils'
> ```

Looks like the issue was that my anaconda install was in /anaconda and therefore required sudo. After reinstalling anaconda in ~/, `--no-build-isolation` is working now.

It would be GREAT to retain ability to build things outside containers!

**Comment by mattsta** [comment]
Seeing `ModuleNotFoundError: No module named 'torch'` during an install is probably because the setup.py is technically incorrect.

python needs more details about dependencies during build time and it's not being threaded through the entire project definition (and it's not great/safe to be calling other installed libraries during install time, etc).

It looks like this is project borrowed a copy of the broken detectron2 setup script which they also haven't fixed for years so the infrastructure contagion is spreading 🫠 

also see:

- https://github.com/python-poetry/poetry/issues/2113
- https://github.com/python-poetry/poetry/issues/2113#issuecomment-1145937065
- https://github.com/python-poetry/poetry/issues/2113#issuecomment-1221601986

**Comment by tridao** [comment]
I had to remove pyproject.toml for now since I couldn't find a way to add torch as a build dependencies that work for everyone. Hopefully installation would work for the newest version (1.0.8).

**Comment by zerogerc** [comment]
Hi, the installation doesn't work for me with 1.0.8. Same error

```
 ModuleNotFoundError: No module named 'torch'
```

**Comment by kylematoba** [comment]
`--no-build-isolation` works for me with 1.0.8, torch==2.0.0+cu117.

**Comment by jaraco** [comment]
This issue is a dupe of #246.

**Comment by albertotono** [comment]
I tried `pip install flash-attn --no-build-isolation` , it did not work for me

![Screenshot 2024-11-08 at 3 22 49 PM](https://github.com/user-attachments/assets/fefce586-5586-4121-80aa-6e1faa6ad0fb)

I am on torch 2.1 and cuda 12.2

**Comment by BossLevel** [comment]
> This issue is a dupe of [#246](https://github.com/Dao-AILab/flash-attention/issues/246).

Agreed - can you remove this one, it's confusing. The only solution that worked for me: https://github.com/Dao-AILab/flash-attention/issues/246#issuecomment-3538523587
